test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -385,6 +386,13 @@ object Assertion extends AssertionVariants {
       BoolAlgebraM.success
     )
 
+  /**
+   * Makes a new assertion that requires an Iterable to have all of the elements
+   * as the other Iterable, though not necessarily in the same order
+   */
+  def hasAllOf[A](other: Iterable[A]): Assertion[Iterable[A]] =
```

Comment:
I'm still a little confused on the contract for this. Is it just supposed to be a version of `hasSameElements` that ignores the number of times an element appears? I think we may also need to think about the name as I confused it with `hasSubset`.

Reply:
Yes, that is correct. Perhaps we can use the name `hasSameDistinctElements `? The name is a bit longer but I think it is more descriptive. The other option is that we simply remove this assertion if there is going to be confusion on the behaviour of these assertions.

Reply:
How about `hasSameElementsDistinct`? That will be more discoverable by IDEs.

Reply:
I have changed the assertion to `hasSameElementsDistinct`. Please let me know if there's anything I need to do before this can be merged into master.